### ISSUE TITLE
Hook in a spell-checking CI

### ISSUE THREAD

**Issue by n8willis** [n8willis/opentype-shaping-documents#133]
Open for suggestions to a good spell-checker / grammar-checker we could hook into in CI fashion.

I used to do that sort of thing locally back when I first started the project, but that was a different machine and, between that and just sort of having a different editing environment set up in other ways, I haven't reconfigured it. So rather than just repeat the one-machine-only model, I would like to toss out the idea of integrating it into a CI workflow.

There are certainly a few options to look at, but I don't have experience with any of them. If anybody does (and has a strong up or down vote about a particular option), please feel free to say.

Certainly the top two issues from my POV are that 
(a) we would need a lot of custom dictionary entries, so if those are a pain to wrangle, it's a minus.
(b) although the nominal language of the docs (at present) is English, simply by virtue of the subject matter the text incorporates lots of non-English content. 

Item b is probably more of a concern for any tool that claims to do grammar checks, but it's certainly a big deal even in adding custom dictionary words.

I'm going to poke around at some options, but in the mean time I'd be happy to hear from others who have experience....

**Comment by alfiedotwtf** [comment]
I think this is going to be a hard one since b)

What would be the aims of the hook i.e reject the PR and make it unmergable until a correction was made, or just a daily/weekly report? I think if you started with aims, it could help with the decision making.

Another thing to make it less laborious could be to run them locally with whatever you're currently using with day-to-day documents, and then declare them as "clean". Then you could turn a spell checker for only PR diffs... that way, it's not constantly rejecting things that already exist in the documents.

**Comment by n8willis** [comment]
> What would be the aims of the hook i.e reject the PR and make it unmergable until a correction was made, or just a daily/weekly report? I think if you started with aims, it could help with the decision making.

Right. Longer-term, my interest is mainly to enforce (either hard or soft enforcement; I don't know which) some standardization on terminology where there is not strictly a universally correct answer, like "codepoint" vs "code point", and ensuring some consistent markup, like wrapping feature tags and other tokens in backticks.

> Another thing to make it less laborious could be to run them locally with whatever you're currently using with day-to-day documents, and then declare them as "clean". Then you could turn a spell checker for only PR diffs... that way, it's not constantly rejecting things that already exist in the documents.

Yeah; I will get something set up again locally. I have historically done a lot of that sort of thing in an Emacs-specific way, though (where there are plenty of options); so when I started thinking about integrating it with CI as well, that added another bullet point to compare them on. I'd rather not maintain two custom-dictionary files in distinct formats and sync them (although ... maybe that's just one more hook to add....).
